iepy/preprocess/ner/regexp.py

```
@@ -0,0 +1,107 @@
+
+    def run_ner(self, doc):
+        entities = []
+        tokens = doc.tokens
+        searcher = TokenSearcher(tokens)
+        for match in searcher.finditer(self.regexp):
+            entity_oc = self.process_match(match)
+            if type(entity_oc) == list:
+                entities.extend(entity_oc)
+            else:
+                entities.append(entity_oc)
+        return entities
+
+    def process_match(self, match):
+        name = ' '.join(match.group())
+        kind = self.label
+        offset, offset_end = match.span()
+        entity_oc = self.build_occurrence(name, kind, name, offset, offset_end)
+
+        return entity_oc
+
+
+class TokenSearcher(NLTKTokenSearcher):
+    """
+    From nltk.text.TokenSearcher documentation:
+
+        A class that makes it easier to use regular expressions to search
+        over tokenized strings.  The tokenized string is converted to a
+        string where tokens are marked with angle brackets -- e.g.,
+        ``'<the><window><is><still><open>'``.  The regular expression
+        passed to the ``findall()`` method is modified to treat angle
+        brackets as non-capturing parentheses, in addition to matching the
+        token boundaries; and to have ``'.'`` not match the angle brackets.
+    """
+
+    def __init__(self, tokens):
+        # replace < and > inside tokens with \< and \>
+        _raw = '><'.join(w.replace('<', '\<').replace('>', '\>') for w in tokens)
+        # preprend >< instead of < for easier token counting
+        self._raw = '><' + _raw + '>'
+
+    def finditer(self, regexp):
+        regexp = preprocess_regexp(regexp)
+
+        i = re.finditer(regexp, self._raw)
+        while True:
+            try:
+                m = next(i)
+                start, end = m.span()
+                token_start = self._raw[:start].count('><')
+                token_end = self._raw[:end].count('><')
+                yield MatchObject(m, token_start, token_end)
+            except:
```

This `try...except` is dangerous because it silently hides any error that can happen inside the loop.

Why not to use `for m in i:` instead?